Dockerfile

```diff
@@ -1,21 +1,40 @@
-FROM node:10.13.0
+FROM node:10.13-alpine
 
-# The key bits here are making sure we install, libsecret-1-dev libglib2.0-dev
-RUN apt-get update -qq && apt-get install -y \
-  libsecret-1-dev libglib2.0-dev && \
-  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+# Install required libraries
+RUN apk --no-cache add bash curl git \
+  make gcc g++ python \
+  libsecret-dev glib-dev
 
+# Set up deploy user and working directory
+RUN adduser -D -g '' deploy
+RUN mkdir -p /app
+RUN chown deploy:deploy /app
+
+# Set up dumb-init
+ADD https://github.com/Yelp/dumb-init/releases/download/v1.2.2/dumb-init_1.2.2_amd64 /usr/local/bin/dumb-init
+RUN chown deploy:deploy /usr/local/bin/dumb-init
+RUN chmod +x /usr/local/bin/dumb-init
+
+# Copy required libs and files to /app
 ADD . /app
+ADD package.json /app
+ADD yarn.lock /app
 WORKDIR /app
 
-RUN rm -f /usr/local/bin/yarn && \
-  curl -o- -L https://yarnpkg.com/install.sh | bash && \
-  chmod +x ~/.yarn/bin/yarn && \
-  ln -s ~/.yarn/bin/yarn /usr/local/bin/yarn
+# Set up node_modules
+RUN yarn cache clean && yarn install
 
-RUN yarn install
+# Copy over node_modules
+ADD --chown=deploy:deploy . /app
```

Comment:
Would it be possible to consolidate _all_ copy behavior behind this single line? Line 20 and 21 are copying files that would naturally be covered by `. /app`.

Reply:
Tried consolidating a few different ways, but never had the right packages -- this was the source of the issues today with pug/loaders.

Reply:
Perhaps a small comment would be good to add so that we don't run into the same issue in the future when another dev comes along and is like "these can be consolidated".

Reply:
I did add comments to both blocks for why we add /app, you find them unclear?

Reply:
If I understand correctly, the earlier `ADD`s of `package.json` and `yarn.lock` happen there because they're necessary for the `yarn install` step that follows them. This leverages smart docker caching so that, e.g., when a random file in `./` changes, it reuses the cached state from _after_ the yarn install step and only repeats the copy to `/app`. If you consolidated the `ADD` steps, _any_ file change would invalidate docker's cache and require a repeat `yarn install`.

Reply:
Re comment, I mean a note about the problem faced earlier, and why they're needed, which was elaborated on just above. It's not entirely intuitive.